Backend/app/rider-platform/rider-app/Main/src/SharedLogic/Scheduler/Jobs/ExecutePaymentIntent.hs

```diff
@@ -0,0 +1,129 @@
+import qualified Domain.Types.Ride as DRide
+import Kernel.Beam.Functions
+import Kernel.External.Encryption (decrypt)
+import Kernel.External.Payment.Interface.Types as Payment
+import Kernel.External.Types (SchedulerFlow)
+import Kernel.Prelude
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified Lib.Payment.Storage.Queries.PaymentOrder as QOrder
+import Lib.Scheduler
+import qualified SharedLogic.CallBPPInternal as CallBPPInternal
+import SharedLogic.JobScheduler
+import SharedLogic.Payment as SPayment
+import qualified Storage.CachedQueries.Merchant as CQM
+import qualified Storage.CachedQueries.Merchant.MerchantOperatingCity as CQMOC
+import qualified Storage.Queries.Booking as QRB
+import qualified Storage.Queries.Person as QPerson
+import qualified Storage.Queries.Ride as QRide
+import Tools.Error
+
+sendExecutePaymentIntent ::
+  ( EncFlow m r,
+    CacheFlow m r,
+    MonadFlow m,
+    EsqDBFlow m r,
+    SchedulerFlow r,
+    HasShortDurationRetryCfg r c
+  ) =>
+  Job 'ExecutePaymentIntent ->
+  m ExecutionResult
+sendExecutePaymentIntent Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  let jobData = jobInfo.jobData
+      personId = jobData.personId
+      rideId = jobData.rideId
+      fare = jobData.fare
+      applicationFeeAmount = jobData.applicationFeeAmount
+  Redis.withWaitOnLockRedisWithExpiry paymentJobExecLockKey 60 60 $ do
+    logDebug "Executing payment intent"
+    QRide.markPaymentStatus DRide.Initiated rideId
+    person <- runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+    ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideNotFound rideId.getId)
+    fareWithTip <- case ride.tipAmount of
+      Nothing -> return fare
+      Just tipAmount -> fare `addPrice` tipAmount
+    customerPaymentId <- person.customerPaymentId & fromMaybeM (PersonFieldNotPresent "customerPaymentId")
+    paymentMethodId <- person.defaultPaymentMethodId & fromMaybeM (PersonFieldNotPresent "defaultPaymentMethodId")
+    driverAccountId <- ride.driverAccountId & fromMaybeM (RideFieldNotPresent "driverAccountId")
+    email <- mapM decrypt person.email
+    let createPaymentIntentReq =
+          Payment.CreatePaymentIntentReq
+            { amount = fareWithTip.amount,
+              applicationFeeAmount,
+              currency = fareWithTip.currency,
+              customer = customerPaymentId,
+              paymentMethod = paymentMethodId,
+              receiptEmail = email,
+              driverAccountId
+            }
+
+    booking <- runInReplica $ QRB.findById ride.bookingId >>= fromMaybeM (BookingDoesNotExist ride.bookingId.getId)
+    paymentIntentResp <- SPayment.makePaymentIntent person.merchantId person.merchantOperatingCityId person.id ride createPaymentIntentReq
+    void $ SPayment.chargePaymentIntent booking.merchantId booking.merchantOperatingCityId paymentIntentResp.paymentIntentId
```

Comment:
Update `chargePaymentIntent` to return charge status and then update the payment status accordingly